bufferserver/src/main/java/com/datatorrent/bufferserver/server/Server.java

```diff
@@ -199,6 +199,13 @@ private void handlePurgeRequest(PurgeRequestTuple request, final AbstractLengthP
     }
   }
 
+  public void purge(long windowId)
+  {
+    for (DataList dataList: publisherBuffers.values()) {
```

Comment:
It is not thread safe to access HashMap (publisherBuffers) from 2 different threads in a case when one thread modifies HashMap (adds or removes entries in the HashMap). It is not fine as there is a race condition between purge() and handlePublisherRequest().